llvm/lib/Target/AArch64/AArch64FrameLowering.cpp

```diff
@@ -5074,7 +5083,11 @@ void AArch64FrameLowering::orderFrameObjects(
       if (AFI.hasStackHazardSlotIndex()) {
         std::optional<int> FI = getLdStFrameID(MI, MFI);
         if (FI && *FI >= 0 && *FI < (int)FrameObjects.size()) {
-          if (MFI.getStackID(*FI) == TargetStackID::ScalableVector ||
+          // Note: PPR accesses should be treated as GPR accesses for the
+          // purposes of stack hazards as PPR ld/sts are handled on the CPU.
+          if ((MFI.getStackID(*FI) == TargetStackID::ScalableVector &&
+               (ZPRPPRSPills ||
```

Comment:
When I remove `ZPRPPRSpills ||` then all tests still pass. Is this condition relevant, or is there missing test-coverage?